lib/nostrum/struct/channel.ex

```diff
@@ -240,47 +575,107 @@ defmodule Nostrum.Struct.Channel do
           name: name
         }
   @typedoc """
-  A `Nostrum.Struct.Channel` that represents a channel in a guild
+  Guild channel types
   """
+  @typedoc deprecated: "See t.0"
   @type guild_channel ::
           guild_text_channel
           | guild_voice_channel
-          | channel_category
+          | guild_category_channel
 
   @typedoc """
-  A `Nostrum.Struct.Channel` that represents a text channel
+  All valid text channels.
   """
+  @typedoc deprecated: "See t.0"
```

Comment:
Hmm. Why would we deprecate these two? I think for functions it makes sense to have the ability to spec that they want a text channel, a voice channel, or a guild channel instead of "any channel".

Reply:
My reasoning was based on where these types are currently used throughout the API and their increasing rate of inaccuracy over time. Searching for any of these groupings is likely to take you to something out of date.

Mostly they are referenced in the documentation for where options are applicable to be used. such as: https://github.com/Kraigie/nostrum/blob/81c36cdceb4ac50ba7e19b39ec0f4489ab8fa975/lib/nostrum/api.ex#L648

One of the new channel types being added is the `guild_news_channel` - this is definitely a text channel. Yet if we add that type of channel to the `text_channel` grouping and follow the current pattern where these types are used. it causes `rate_limit_per_user` to now be lying to us in the documentation because it is not applicable. So we could amend the `rate_limit_per_user` to say that it is valid for all `t:text_channels/0` except `t:guild_news_channel/0`. This seemed like it would quickly get out of control and be a pretty big job and an even bigger job to keep up to date.

As a solution - I considered rewriting the groupings per the `CHANNEL TYPE` in this table: https://discord.com/developers/docs/resources/channel#modify-channel-json-params-guild-channel & https://discord.com/developers/docs/resources/channel#channel-object-channel-types

Which would equate to new categories of : 


**TEXT_CHANNEL** 
- `GUILD_TEXT`
- `DM`
- `GROUP_DM`
- `GUILD_NEWS`
- `GUILD_NEWS_THREAD`
- `GUILD_PUBLIC_THREAD`
- `GUILD_PRIVATE_THREAD`

**GUILD_CHANNEL**
- `GUILD_TEXT`
- `GUILD_NEWS`
- `GUILD_NEWS_THREAD`
- `GUILD_PUBLIC_THREAD`
- `GUILD_PRIVATE_THREAD`
- `GUILD_VOICE`
- `GUILD_CATEGORY`
- `GUILD_STORE`
- `GUILD_STAGE_VOICE`

**THREAD_CHANNEL**
- `GUILD_NEWS_THREAD`
- `GUILD_PUBLIC_THREAD`
- `GUILD_PRIVATE_THREAD`

**VOICE CHANNEL**
- `GUILD_VOICE`
- `GUILD_STAGE_VOICE `

But doing that in this PR and not going through the API documentation and updating their referenced locations did not seem appropriate either. 

My conclusion was that given the rate of change with the discord API, I don't think these loose arbitrary groupings are doing anybody any favours by sticking around.